crates/wasmparser/src/readers/core/coredumps.rs

```diff
@@ -0,0 +1,150 @@
+use crate::{BinaryReader, FromReader, Result};
+
+/// The data portion of a custom section representing a core dump. Per the
+/// tool-conventions repo, this section just specifies the executable name that
+/// the core dump came from while the rest of the core dump information is
+/// contained in a corestack custom section
+///
+/// # Examples
+///
+/// ```
+/// use wasmparser::{ BinaryReader, CoreDumpSection, FromReader, Result };
+/// let data: &[u8] = &[0x00, 0x09, 0x74, 0x65, 0x73, 0x74, 0x2e, 0x77, 0x61,
+///      0x73, 0x6d];
+/// let mut reader = BinaryReader::new(data);
+/// let core = CoreDumpSection::from_reader(&mut reader).unwrap();
+/// assert!(core.name == "test.wasm")
+/// ```
+pub struct CoreDumpSection<'a> {
+    /// The name of the process that created the core dump
+    pub name: &'a str,
+}
+
+impl<'a> FromReader<'a> for CoreDumpSection<'a> {
+    fn from_reader(reader: &mut BinaryReader<'a>) -> Result<Self> {
+        let pos = reader.original_position();
+        if reader.read_u8()? != 0 {
+            bail!(pos, "invalid start byte for core dump name");
+        }
+        let name = reader.read_string()?;
+        Ok(CoreDumpSection { name })
+    }
+}
+
+/// The data portion of a custom section representing a core dump stack. The
+/// structure of this follows the coredump spec in the tool-conventions repo
+///
+/// # Examples
+///
+/// ```
+/// let data: &[u8] = &[0x00, 0x04, 0x6d, 0x61, 0x69, 0x6e, 0x01, 0x00, 0x2a,
+///     0x33, 0x01, 0x7f, 0x01, 0x01, 0x7f, 0x02];
+/// use wasmparser::{ BinaryReader, CoreDumpStackSection, FromReader };
+/// let mut reader = BinaryReader::new(data);
+/// let corestack = CoreDumpStackSection::from_reader(&mut reader).unwrap();
+/// assert!(corestack.name == "main");
+/// assert!(corestack.frames.len() == 1);
+/// let frame = &corestack.frames[0];
+/// assert!(frame.funcidx == 42);
+/// assert!(frame.codeoffset == 51);
+/// assert!(frame.locals.len() == 1);
+/// assert!(frame.stack.len() == 1);
+/// ```
+pub struct CoreDumpStackSection<'a> {
+    /// The thread name
+    pub name: &'a str,
+    /// The stack frames for the core dump
+    pub frames: Vec<CoreDumpStackFrame>,
```

Comment:
We might eventually want to split these out into a sub-reader and not eagerly parse and allocate a `Vec` for them, but this is fine for now.